lints/lint_subject_contains_reserved_arpa_ip.go

```diff
@@ -0,0 +1,232 @@
+/*
+ * ZLint Copyright 2019 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package lints
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+)
+
+const (
+	// arpaTLD holds a string constant for the .arpa TLD
+	arpaTLD = ".arpa"
+
+	// rdnsIPv4Suffix is the expected suffix for IPv4 reverse DNS names as
+	// specified in https://tools.ietf.org/html/rfc1035#section-3.5
+	rdnsIPv4Suffix = ".in-addr" + arpaTLD
+	// rndsIPv4Labels is the expected number of labels for an IPv4 reverse DNS
+	// name (not counting the rdnsIPv4Suffix labels). IPv4 addresses are four
+	// bytes. RFC 1035 uses one byte per label meaning there are 4 expected labels
+	// under the rdnsIPv4Suffix.
+	rdnsIPv4Labels = 4
+
+	// rdnsIPv6Suffix is the expected suffix for IPv6 reverse DNS names as
+	// specified in https://tools.ietf.org/html/rfc3596#section-2.5
+	rdnsIPv6Suffix = ".ip6" + arpaTLD
+	// rndsIPv6Labels is the expected number of labels for an IPv6 reverse DNS
+	// name (not counting the rdnsIPv6Suffix labels). IPv6 addresses are 16 bytes.
+	// RFC 3596 Sec 2.5 uses one *nibble* per label meaning there are 16*2
+	// expected labels under the rdnsIPv6Suffix.
+	rdnsIPv6Labels = 32
+)
+
+// arpaReservedIP is a linter that errors for any well formed rDNS names in the
+// .in-addr.arpa or .ip6.arpa zones that specify an address in an IANA reserved
+// network.
+// See also: lint_subject_contains_malformed_arpa_ip.go for a lint that warns
+// about malformed rDNS names in these zones.
+type arpaReservedIP struct{}
+
+// Initialize for an arpaReservedIP linter is a NOP to statisfy linting
+// interfaces.
+func (l *arpaReservedIP) Initialize() error {
+	return nil
+}
+
+// CheckApplies returns true if the certificate contains any names that end in
+// one of the two designated zones for reverse DNS: in-addr.arpa or ipv6.arpa.
```

Comment:
Comment should reference ip6.arpa. There's no **v** in the zone name. See [RIPE](https://www.ripe.net/manage-ips-and-asns/db/support/documentation/glossary/ip6.arpa) .